### PR TITLE
[Spring]Add support for byos subPath

### PR DESCRIPTION
**Pull request by Descatles**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Spring

**Comment by zhoxing-ms** [comment]
Could you please add some test case for this new parameter?

File: src/spring/HISTORY.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 Release History
 ===============
+1.9.1
+---
+* Support subPath for bring your own persistent storage feature.
+* Add new parameter `--enable-sub-path` into `az spring append-persistent-storage` to enable subPath feature.
```
Comment (zhoxing-ms) [inline src/spring/HISTORY.md:6]: If you want to release a new extension version for this PR, please also upgrade the version defined in the `setup.py`

**Comment by Descatles** [comment]
> Could you please add some test case for this new parameter?

This cli command's test is not in this cli repo but in our end to end test repo. It will be covered by our end to end test

**Comment by Descatles** [comment]
Remember to add test for az spring append-persistent-storage in the future

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**